### DISCUSSION TITLE
Getting a error when I use flatMap() and merge() with string keys

### DISCUSSION THREAD

**Discussion by poldixd** [larastan/larastan#1427] [Q&A]
Hi there,

I'm trying to create a dynamic validation array for Livewires `$this->validate`.  The result should look like this:

```php
[
    'name_de' => 'required|max:191',
    'description_de' => 'max:65535',
    'name_en' => 'required|max:191',
    'description_en' => 'max:65535',
    'latitude' => 'required|numeric',
    'longitude' => 'required|numeric',
]
```

This is my code:

```php
collect(['de', 'en'])
    ->flatMap(fn ($localeCode) => [
        "name_{$localeCode}" => 'required|max:191',
        "description_{$localeCode}" => 'max:65535',
    ])
    ->merge([
        'latitude' => 'required|numeric',
        'longitude' => 'required|numeric',
    ]);
```

When I run `vendor/bin/phpstan analyse` I'm getting this error:

```
Parameter #1 $items of method Illuminate\Support\Collection<int,mixed>::merge() expects Illuminate\Contracts\Support\Arrayable<int, mixed>|iterable<int, mixed>, array{latitude: 'required|numeric', longitude: 'required|numeric'} given. 
```

Maybe I have to set manually the types?

My `phpstan.neon` is:

```yml
includes:
    - ./vendor/nunomaduro/larastan/extension.neon

parameters:

    paths:
        - app/

    level: 5
```

**Comment by szepeviktor** [comment]
Hello @poldixd!

I think it is about array keys not being integers.

**Reply by poldixd** [reply]
Hey @szepeviktor,

thanks for your answer. If I use `mapWithKeys()` there is no error:

```php
collect(['de', 'en'])
    ->mapWithKeys(fn ($localeCode) => ["description_{$localeCode}" => 'max:65535'])
    ->merge([
        'latitude' => 'required|numeric',
        'longitude' => 'required|numeric',
    ]);
```

```
[OK] No errors  
```

**Comment by mad-briller** [comment]
the collection generics are controlled directly by Laravel, not Larastan:
https://github.com/laravel/framework/blob/9.x/src/Illuminate/Collections/Traits/EnumeratesValues.php#L384-L393

as you can see, the `flatMap` method is phpdoc'd to return a `static<int, mixed>` directly, throwing away the generics
this is probably better off reported to Laravel. 

If they don't want to support it, then perhaps Larastan can add a stub to provide the information

**Reply by mad-briller** [reply]
https://github.com/laravel/framework/pull/44772
